### PR TITLE
Refactor client types and options

### PR DESCRIPTION
**Pull request by ericHgorski**
This commit refactors the client types and options in the `client-types.ts` file. It introduces new interfaces for common types, signer getters, gas options, transaction callbacks, skip client options, execute route options, execute cosmos message options, and sign cosmos message options.

### CODE REVIEW

**Comment by ericHgorski** [comment]
Noticed alot of repeat and unified typing from my docs work. Attempted to clean it up. This may be a hazardous PR to merge mid-widget v2 sprint however.

File: packages/client/src/types/unified.ts [hunk 1]
```diff
@@ -550,6 +550,8 @@ export type MsgsResponse = {
 
 export type BridgeType = 'IBC' | 'AXELAR' | 'CCTP' | 'HYPERLANE' | 'OPINIT';
 
+export type ChainType = 'cosmos' | 'evm' | 'svm' | 'sei';
```
Comment (codingki) [inline packages/client/src/types/unified.ts:553]: sei is not a chain type, please remove sei

Comment (ericHgorski) [reply packages/client/src/types/unified.ts:553]: we treat it like a chainType here. should this be refactored? 

https://github.com/skip-mev/skip-go/blob/4629bcaeadb3430d3e722f393638b254c74dcbc8/packages/widget-v2/src/hooks/useWalletList.tsx#L25

Comment (ericHgorski) [reply packages/client/src/types/unified.ts]: ok resolved this here: https://github.com/skip-mev/skip-go/pull/334/commits/3bd27716e42c64c5fc6f3c6e7e3402ab4144f40e

**Review by codingki — APPROVED** [review]
LGTM

**Pull request merged**